test/parallel/test-whatwg-encoding.js

```diff
@@ -0,0 +1,94 @@
+// Flags: --expose-internals
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { Buffer, TextDecoder, TextEncoder } = require('buffer');
```

Comment:
I'm guessing here: tests are supposed to be self-contained scripts, while everything in `lib` is user-visible and -contaminatable, like a seemingly innocent `var Buffer = {}` line in REPL. See https://github.com/nodejs/node-convergence-archive/issues/21.